NadekoBot.Core/Modules/Searches/Services/SearchesService.cs

```diff
@@ -360,13 +361,27 @@ public string GetRandomImageUrl(ImageTag tag)
             }
             else if (tag == ImageTag.Goose){
                 return $"https://raw.githubusercontent.com/steggie3/goose-dataset/master/images/goose-mugshot-{_rng.Next(1, max).ToString("0000")}.jpg";
-            }else{
+            }
+            else if (tag == ImageTag.Image){
+                return $"https://picsum.photos/seed/{GetRandomString(16)}/2000";
+            }
+            else{
  
             return $"https://nadeko-pictures.nyc3.digitaloceanspaces.com/{subpath}/" +
                 _rng.Next(1, max).ToString("000") + ".png";
             }
         }
 
+        private string GetRandomString(int length) {
+            const int aInASCII = 97;
+            const int zInASCII = 122;
+            var randomString = "";
+            for (var i = 0; i < length; i++) {
+                randomString += (char)(_rng.Next(aInASCII, zInASCII));
+            }
```

Comment:
For a tiny string like this it doesn't matter but in general you should be wary of doing string concatenation in a loop. Without optimizations, it will take time quadratic in the length of the string. But string formatting and concatenation have different idioms in every language so you'll just have to be aware of it in every language you work with.

An alternative solution would be something like https://stackoverflow.com/a/1344242

Reply:
Yeah I just looked up ways of generating random strings in C# and picked the one I understood best. I'll change it to the one you suggested